boards/arm/rp2040/common/src/rp2040_common_bringup.c

```diff
@@ -68,6 +68,11 @@
 #include "rp2040_bmp280.h"
 #endif
 
+#ifdef CONFIG_SENSORS_SHT4X
+#include <nuttx/sensors/sht4x.h>
+#include "rp2040_i2c.h"
```

Comment:
I think you don't need to include i2c header file here. Modify sht4x_register() to accept the busno instead of passing the i2c_master struct pointer to it.